lib/application_insights/channel/contracts/data.rb

```diff
@@ -1,48 +1,9 @@
 require_relative 'json_serializable'
 
-module ApplicationInsights
-  module Channel
-    module Contracts
-      # Data contract class for type Data.
-      class Data < JsonSerializable
-        # Initializes a new instance of the Data class.
-        def initialize(options={})
-          defaults = {
-            'baseType' => nil,
-            'baseData' => nil
-          }
-          values = {
-            'baseData' => nil
-          }
-          super defaults, values, options
-        end
-        
-        # Gets the base_type property.
-        def base_type
-          @values.fetch('baseType') { 
-            @values['baseType'] = nil
-          }
-        end
-        
-        # Sets the base_type property.
-        def base_type=(value)
-          if value == @defaults['baseType']
-            @values.delete 'baseType' if @values.key? 'baseType'
-          else
-            @values['baseType'] = value
-          end
-        end
-        
-        # Gets the base_data property.
-        def base_data
-          @values['baseData']
-        end
-        
-        # Sets the base_data property.
-        def base_data=(value)
-          @values['baseData'] = value
-        end
-      end
-    end
+module ApplicationInsights::Channel::Contracts
+  class Data
+    include JsonSerializable
+
+    attributes :baseType, :baseData
```

Comment:
I will implement the latest proposal: the one with `json_mapping` method (btw if you have a better name please help me :) )

Anyways I have to insist on using the following syntax, because is more descriptive and it is really common in ruby-land. Most of the ORMs uses this syntax.

```ruby
module ApplicationInsights::Channel::Contracts
  class User
    include JsonSerializable

    attribute :account_acquisition_date, name: 'ai.user.accountAcquisitionDate'
    attribute :account_id, name: 'ai.user.accountId'
    attribute :user_agent, name: 'ai.user.userAgent'
    attribute :id, name: 'ai.user.userAgent', default: 1
  end
end
```

If you change your mind and accept this _ORM-like_ syntax, I'm glad to update the PullRequest.